src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -186,6 +218,18 @@ namespace AppInstaller::Repository
                 // Lie here so that list and upgrade will carry on to be able to output the diagnostic information.
                 return true;
             }
+
+            bool IsSame(const IPackage* other) const override

```

Comment:
>IsSame [](start = 17, length = 6)

Should UnknownPackage != UnknownPackage? Or there could only be one UnknownPackage?

Reply:
The goal is that two of the same searches should return results with IsSame being 1:1 correlated.  A null != null kind of behavior would make that fail.